eventstore/dynamo/dynamo.go

```diff
@@ -0,0 +1,193 @@
+package dynamo
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+	"github.com/hallgren/eventsourcing/core"
+)
+
+type DynamoEventStore struct {
+	db        *dynamodb.DynamoDB
+	tableName string
+}
+
+func New(sess *session.Session, tableName string) *DynamoEventStore {
+	return &DynamoEventStore{
+		db:        dynamodb.New(sess),
+		tableName: tableName,
+	}
+}
+
+func (store *DynamoEventStore) Save(events []core.Event) error {
+	// Comprobar si hay eventos
+	if len(events) == 0 {
+		return nil
+	}
+
+	lastVersion, err := store.getLastVersion(events[0].AggregateID)
+	if err != nil {
+		// Manejar error al obtener la última versión
+		return err
+	}
+
+	// Comprobar si la versión del evento es la siguiente versión esperada
+	if core.Version(lastVersion)+1 != events[0].Version {
+		return core.ErrConcurrency
+	}
+
+	for i := range events {
+		// Obtener y actualizar el contador global de versión para cada evento
+		globalVersion, err := store.getAndUpdateGlobalVersion()
+		if err != nil {
+			return err
+		}
+
+		// Establecer la versión global del evento
+		events[i].GlobalVersion = globalVersion
+
+		av, err := dynamodbattribute.MarshalMap(events[i])
+		if err != nil {
+			return err
+		}
+
+		// Preparar la expresión de condición para control de concurrencia
+		condition := "attribute_not_exists(AggregateID) OR Version = :version"
+
+		av["Version"] = &dynamodb.AttributeValue{
+			N: aws.String(fmt.Sprintf("%d", events[i].Version)),
+		}
+		av["GlobalVersion"] = &dynamodb.AttributeValue{
+			N: aws.String(fmt.Sprintf("%d", events[i].GlobalVersion)),
+		}
+
+		input := &dynamodb.PutItemInput{
+			TableName: aws.String(store.tableName),
+			Item:      av,
+			ExpressionAttributeValues: map[string]*dynamodb.AttributeValue{
+				":version": {N: aws.String(fmt.Sprintf("%d", events[i].Version-1))},
+			},
+			ConditionExpression: aws.String(condition),
+		}
+
+		_, err = store.db.PutItem(input)
+		if err != nil {
+			switch err.(type) {
+			case *dynamodb.ConditionalCheckFailedException:
+				return core.ErrConcurrency
+			}
+
+			return err
+		}
+
+	}
+	return nil
+}
+
+func (store *DynamoEventStore) Get(
+	ctx context.Context,
+	id string,
+	aggregateType string,
+	afterVersion core.Version,
+) (core.Iterator, error) {
+	queryInput := &dynamodb.QueryInput{
+		TableName:              aws.String(store.tableName),
```

Comment:
In these cases it is possible to simply give it a reference to the string:
```suggestion
		TableName:              &store.tableName,
```